File: praktikumsplaner-frontend/src/types/Ausbildungsrichtung.ts
export const Ausbildungsrichtung = [
    {
        name: "Fachinformatiker Systemintegration (FISI)",
        value: "FISI",
        color: "light-blue-lighten-3",
    },
    {
        name: "Verwaltungswirt*in (QE2)",
        value: "QE2",
        color: "teal-lighten-3",
    },
    {
        name: "Kauffrau und Kaufmann für Büromanagement (KfB)",
        value: "KFB",
        color: "lime-lighten-3",
    },
    {
        name: "Verwaltungsfachangestellte*r (VFAK)",
        value: "VFAK",
        color: "orange-lighten-3",
    },
] as const;

export function findAusbildungsrichtungColorByValue(value: string): string {
    const ausbildungsrichtung = Ausbildungsrichtung.find(
        (s) => s.value === value
    );
    return ausbildungsrichtung ? ausbildungsrichtung.color : "";
}

File: praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/domain/enums/Ausbildungsrichtung.java
package de.muenchen.oss.praktikumsplaner.domain.enums;

import lombok.Getter;

@Getter
public enum Ausbildungsrichtung {
    FISI("Fachinformatiker*in für Systemintegration"),
    QE2("Verwaltungswirt*in"),
    KFB("Kaufleute für Büromanagement"),
    VFAK("Verwaltungsfachangestellte*r kommunal");

    private final String longName;

    Ausbildungsrichtung(final String longName) {
        this.longName = longName;
    }

    public static Ausbildungsrichtung fromBildungsrichtung(final Bildungsrichtung bildungsrichtung) {
        if (bildungsrichtung.getArt() != Bildungsrichtung.Art.AUSBILDUNG) {
            throw new IllegalArgumentException("Bildungsrichtung " + bildungsrichtung + " ist keine Ausbildungsrichtung");
        }
        return Ausbildungsrichtung.valueOf(bildungsrichtung.name());
    }
}
